src/CommunityToolkit.Maui.Core/Views/DrawingView/PlatformView/MauiDrawingView.shared.cs

```diff
@@ -0,0 +1,248 @@
+			{
+				new(previousPoint.X, previousPoint.Y)
+			},
+			LineColor = LineColor,
+			LineWidth = LineWidth
+		};
+
+		Redraw();
+
+		Lines.CollectionChanged += OnLinesCollectionChanged;
+	}
+
+	void OnMoving(PointF currentPoint)
+	{
+		if (!isDrawing)
+		{
+			return;
+		}
+
+#if !ANDROID
+		AddPointToPath(currentPoint);
+#endif
+
+		Redraw();
+		currentLine?.Points.Add(currentPoint);
+	}
+
+	void OnFinish()
+	{
+		if (currentLine is not null)
+		{
+			Lines.Add(currentLine);
+			OnDrawingLineCompleted(currentLine);
+		}
+
+		if (ClearOnFinish)
+		{
+			Lines.Clear();
+			ClearPath();
+		}
+
+		currentLine = null;
+		isDrawing = false;
+	}
+
+	void OnCancel()
+	{
+		currentLine = null;
+		ClearPath();
+		Redraw();
+		isDrawing = false;
+	}
+
+	void OnDrawingLineCompleted(MauiDrawingLine lastDrawingLine) =>
+		weakEventManager.HandleEvent(this, new MauiDrawingLineCompletedEventArgs(lastDrawingLine), nameof(DrawingLineCompleted));
+
+	void OnLinesCollectionChanged(object? sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e) => LoadLines();
+
+	void AddPointToPath(PointF currentPoint) => currentPath.LineTo(currentPoint);
+
+	void LoadLines()
+	{
+		ClearPath();
+		Redraw();
+	}
+
+	void ClearPath()
+	{
+		currentPath = new PathF();
+	}
+
+	void Redraw()
+	{
+		Invalidate();
+	}
+
+	class DrawingViewDrawable : IDrawable
+	{
+		readonly MauiDrawingView drawingView;
+
+		public DrawingViewDrawable(MauiDrawingView drawingView)
+		{
+			this.drawingView = drawingView;
+		}
+
+		public void Draw(ICanvas canvas, RectF dirtyRect)
+		{
+			canvas.SetFillPaint(drawingView.Paint, dirtyRect);
+			canvas.FillRectangle(dirtyRect);
+
+			drawingView.DrawAction?.Invoke(canvas, dirtyRect);
+
+			DrawCurrentLines(canvas, drawingView);
+			SetStroke(canvas, drawingView.LineWidth, drawingView.LineColor);
+
+			canvas.DrawPath(drawingView.currentPath);
+		}
+
+		static void SetStroke(in ICanvas canvas, in float lineWidth, in Color lineColor)
+		{
+			canvas.StrokeColor = lineColor;
+			canvas.StrokeSize = lineWidth;
+		}
+
+		static void DrawCurrentLines(in ICanvas canvas, in MauiDrawingView drawingView)
+		{
+			foreach (var line in drawingView.Lines)
+			{
+				var path = new PathF();
+				var points = line.EnableSmoothedPath
+					? line.Points.SmoothedPathWithGranularity(line.Granularity)
+					: line.Points;
+#if ANDROID
+				points = CreateCollectionWithNormalizedPoints(points, drawingView.Width, drawingView.Height);
+#endif
+				if (points.Count > 0)
+				{
+					path.MoveTo(points[0].X, points[0].Y);
+					foreach (var point in points)
+					{
+						path.LineTo(point);
+					}
+
+					SetStroke(canvas, line.LineWidth, line.LineColor);
+					canvas.DrawPath(path);
+				}
+			}
+		}
+
+#if ANDROID
+		static ObservableCollection<PointF> CreateCollectionWithNormalizedPoints(in ObservableCollection<PointF> points, in int drawingViewWidth, in int drawingViewHeight)
+		{
+			var newPoints = new List<PointF>();
```

Comment:
We discussed with Pedro about it. Can we use ObservableCollection collection instead of list to avoid additional allocation?

Or monitoring collection has higher impact?

Reply:
`ObservableCollection` has a bunch of overhead. For example, it fires the `CollectionChanged` event each time `Add`, `Remove` and `Clear` are called.

In general, it should only be used when its `CollectionChanged` or `PropertyChanged` events are required.

I didn't run a Benchmark for this method, but I imagine the allocation of a `List` is fairly minimal.

Reply:
I'll do the benchmark during the code freeze if I've time and will try to optimize it. The ideal would be to use `Span<T>` (when it applies). We can leave as it's for now.